honeybee_schema/radiance/material.py

```diff
@@ -0,0 +1,280 @@
+                    '(default: 0).'
+    )
+
+    g_reflectance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the green channel reflectance '
+                    '(default: 0).'
+    )
+
+    b_reflectance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the blue channel reflectance '
+                    '(default: 0).'
+    )
+
+    specularity: float = Field(
+        default=0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the fraction of specularity. '
+                    'Specularity fractions greater than 0.1 are not realistic. '
+                    '(default: 0).'
+    )
+
+    roughness: float = Field(
+        default=0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the roughness, specified as the '
+                    'rms slope of surface facets. Roughness greater than 0.2 are '
+                    'not realistic (default: 0).'
+    )
+
+
+class Trans(Plastic):
+    """Radiance Translucent material."""
+
+    type: constr(regex='^Trans$') = 'Trans'
+
+    transmitted_diff: float = Field(
+        default=0,
+        description='The fraction of transmitted light that is transmitted diffusely in '
+                    'a scattering fashion (default: 0).'
+    )
+
+    transmitted_spec: float = Field(
+        default=0,
+        description='The fraction of transmitted light that is not diffusely scattered '
+                    '(default: 0).'
+    )
+
+
+class Glass(ModifierBase):
+    """Radiance glass material."""
+
+    type: constr(regex='^Glass$') = 'Glass'
+
+    r_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the red channel transmissivity '
+                    '(default: 0).'
+    )
+
+    g_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the green channel transmissivity '
+                    '(default: 0).'
+    )
+
+    b_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the blue channel transmissivity '
+                    '(default: 0).'
+    )
+
+    refraction_index: float = Field(
+        default=1.52,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the index of refraction '
+                    '(default: 1.52).'
+    )
+
+
+class BSDF(ModifierBase):
+    """Radiance BSDF (Bidirectional Scattering Distribution Function) material."""
+
+    type: constr(regex='^BSDF$') = 'BSDF'
+
+    up_orientation: List[float] = Field(
+        default=(0.01, 0.01, 1.00),
+        description='Vector as sequence that sets the hemisphere that the BSDF material '
+                    'faces. (default: (0.01, 0.01, 1.00).'
+    )
```

Comment:
You need to add the key to ensure the length of items is equal to 3.

Reply:
I can't find a way to fix length in pydantic docs: https://pydantic-docs.helpmanual.io/usage/schema/#field-customisation 

So I did this:
min_items=3,
max_items=3,

Which is kind of ugly, but seems to work.

Reply:
I can confirm that this is the way to do it and I have applied it elsewhere. I know it doesn't feel so elegant at first but, when you think about it, it's kinda following the tenant:
"There should be one, and preferably only one, obvious way to do it."